test/CMakeLists.txt

```diff
@@ -91,40 +91,50 @@ foreach (test_src IN LISTS test_srcs)
   get_filename_component(test_name "${test_src}" NAME_WE)
   string(REGEX REPLACE "^test_" "" test_name "${test_name}")
 
-  # Some tests change behavior based on whether the compiler defs QUICK_TEST
-  # and/or QUICKER_TEST are defined. Detect these and build variants for each
+  # Some tests change behavior based on whether the compiler defs BENCHMARK
+  # and/or MINIMAL_TEST are defined. Detect these and build variants for each
   # configuration:
   cub_check_for_test_variants("${test_src}")
 
   foreach(cub_target IN LISTS CUB_TARGETS)
     if (NO_VARIANTS)
       # Only one version of this test.
       cub_add_test(test_target ${test_name} "${test_src}" ${cub_target})
+    elseif (NOT CUB_TEST_VARIANT STREQUAL "ALL")
+      string(TOLOWER CUB_TEST_VARIANT lowercase_cub_test_variant)
+      cub_add_test(test_target_${lowercase_cub_test_variant}
+        ${test_name}.${lowercase_cub_test_variant}
+        "${test_src}"
+        ${cub_target}
+      )
+      target_compile_definitions(${test_target_${lowercase_cub_test_variant}}
+                                 PRIVATE CUB_TEST_${CUB_TEST_VARIANT})
     else()
       # By default (no flags), the "thorough" version of the test is built:
       cub_add_test(test_target_thorough
         ${test_name}.thorough
         "${test_src}"
         ${cub_target}
       )
+      target_compile_definitions(${test_target_thorough} PRIVATE CUB_TEST_THOROUGH)
```

Comment:
Nothing defines this, is this addition intentional?

Reply:
If we add this here, it should be added to the Makefile, too.

Reply:
Done. I just added it for completeness.